File: src/main/java/com/dafttech/terra/engine/gui/anchors/AnchorRight.java
package com.dafttech.terra.engine.gui.anchors;

import com.dafttech.terra.engine.gui.GUIObject;
import com.dafttech.terra.engine.gui.containers.GUIContainer;

public class AnchorRight extends GUIAnchor {
    public float position = 0;

    public AnchorRight(float p) {
        position = p;
    }

    @Override
    public void applyAnchor(GUIObject object, GUIContainer container) {
        object.position.x = container.size.x * (1f - position) - object.size.x;
    }

}

File: terra/src/main/scala/com/dafttech/terra/engine/gui/anchors/AnchorRight.scala
package com.dafttech.terra.engine.gui.anchors

import com.dafttech.terra.engine.gui.GUIObject
import com.dafttech.terra.engine.gui.containers.GUIContainer

class AnchorRight(val position: Float) extends GUIAnchor {
  def applyAnchor(gObj: GUIObject, container: GUIContainer) {
    gObj.position = gObj.position.withX(container.size.x * (1f - position) - gObj.size.x)
  }
}